assignment/src/models/appointment.ts

```diff
@@ -0,0 +1,59 @@
+import Joi from 'joi';
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface IPatient {
+    _id: mongoose.Types.ObjectId;
+    petName: string;
+    ownerPhoneNumber: string;
+}
+
+interface IAppointment extends Document {
+    patient: IPatient;
+    appointmentStartTime: Date;
+    appointmentEndTime: Date;
+    description: string;
+    feeAmount: number;
+    feeStatus: 'USD' | 'EUR' | 'Bitcoin' | 'Unpaid';
+}
+
+const appointmentSchema = new Schema<IAppointment>({
+    patient: { 
+        type: new Schema({
+            _id: { type: Schema.Types.ObjectId, required: true },
+            petName: { type: String, required: true },
+            ownerPhoneNumber: { type: String, required: true }
+        }),  
+        required: true
+    },
+    appointmentStartTime: { type: Date, required: true },
+    appointmentEndTime: { type: Date, required: true },
+    description: { 
+        type: String, 
+        required: true, 
+        minlength: 5, 
+        maxlength: 255 
+    },
+    feeAmount: { type: Number, required: true },
+    feeStatus: { 
+        type: String, 
+        enum: ['USD', 'EUR', 'Bitcoin', 'Unpaid'], 
+        required: true 
+    }
+});
+
+const Appointment = mongoose.model<IAppointment>('Appointment', appointmentSchema);
+
+function validateAppointment(appointment: any) {
```

Comment:
Please create a folder validations and move your validation functions their
validations/appointment/appointment.ts.

Reason
There can be mutlple validations for a single model